File: BMW-ESM-APP-API/BMW-ESM-APP-API/Models/Entities/Vehicle.cs
using System.ComponentModel.DataAnnotations;

namespace BMW_ESM_APP_API.Models.Entities
{
    public class Vehicle
    {
        [Key]
        public int VehicleID { get; set; }
        public virtual Vehicle_Model VehicleModelID { get; set; }
        public string VIN_Number { get; set; }
        public virtual Warranty_Plan WarrantyPlanID { get; set; }
        public virtual Maintenance_Plan MaintenancePlanID { get; set; }

    }
}

File: src/app/models/Vehicle.ts
export interface Vehicle{
    map(arg0: (e: any) => { id: any; VehicleID: any; VINNumber: any; vehicleModel: any; year: string; }): any[];
    viN_Number: string;
    vehicleModel: string;
    registration: string;
    warrantyPlan: string;
    fleetID: string | null;
}